### PR TITLE
hlint and me: remove redundant brackets

### PR DESCRIPTION
**Pull request by lwm**
Working on something for https://github.com/commercialhaskell/stack/issues/2592.

I'm trying to keep the patches smaller by focusing on a single hlint suggestion at a time. Sorry about the whitespace noise, my `$EDITOR`/`hlint --refactor --refactor-options -i` did it.

### CODE REVIEW

File: src/Stack/Build/ConstructPlan.hs [hunk 1]
```diff
@@ -622,15 +622,15 @@ describeConfigDiff config old new
...
 
 psDirty :: PackageSource -> Maybe (Set FilePath)
 psDirty (PSLocal lp) = lpDirtyFiles lp
-psDirty (PSUpstream {}) = Nothing -- files never change in an upstream package
+psDirty PSUpstream {} = Nothing -- files never change in an upstream package
 
 psLocal :: PackageSource -> Bool
 psLocal (PSLocal _) = True
-psLocal (PSUpstream {}) = False
+psLocal PSUpstream {} = False
```
Comment (Blaisorblade) [inline src/Stack/Build/ConstructPlan.hs]: I... didn't know you could do that. I guess this is fine, calling out in case anybody wants to look at this.

Comment (Blaisorblade) [inline src/Stack/Build/ConstructPlan.hs]: Actually, the later `Cmd{..}` looks fine. I don't know if there's any directive from coding styles here, but I'd say `PSUpstream{}` is clearer than `PSUpstream {}`.

**Comment by Blaisorblade** [comment]
I've looked at all other examples and I had only one comment. LGTM otherwise. Any other opinion?

File: src/Stack/Build/ConstructPlan.hs (hunk 1 above)
Comment (lwm) [inline src/Stack/Build/ConstructPlan.hs]: Fair enough. I poked around and seen more `Foo{}` than `Foo {}`.

**Comment by lwm** [comment]
Oh, I forgot to run this on `test/*`, so added that as well. See small changes to `test/integration/lib/StackTest.hs` and `test/integration/IntegrationSpec.hs`.

**Comment by Blaisorblade** [comment]
I just added a small fixup to **your branch**, will squash-merge that together as soon as Travis accepts this (ahem).

**Comment by lwm** [comment]
:rocket:

**Pull request merged**